tests/test_utils_misc/test_return_with_argument_inside_generator.py

```diff
@@ -0,0 +1,39 @@
+import unittest
+
+from scrapy.utils.misc import is_generator_with_return_value
+
+
+class UtilsMiscPy3TestCase(unittest.TestCase):
+
+    def test_generators_with_return_statements(self):
+        """
+        'return with argument inside generator' is a SyntaxError before Python 3.3
+        """
+        def f():
+            yield 1
+            return 2
+
+        def g():
+            yield 1
+            return 'asdf'
+
+        def h():
+            yield 1
+            return None
+
+        def i():
+            yield 1
+            return
+
+        def j():
+            yield 1
+
+        assert is_generator_with_return_value(f)
+        assert is_generator_with_return_value(g)
+        assert not is_generator_with_return_value(h)
+        assert not is_generator_with_return_value(i)
+        assert not is_generator_with_return_value(j)
+
```

Comment:
I can think of a test case (expected failure?) that we might want to cover here: A test that verifies that, if something like `yield from` is used and this issue affects the nested generator, we will still not detect it because we do not inspect callbacks recursively (which I assume would make things much more complicated and/or have a potentially-significant negative impact on performance).

Reply:
You're absolutely right. The nested generator could even be defined in a different module, it could get messy to get the source code to build the AST.